test/unit/math/mix/scal/fun/lb_constrain_test.cpp

```diff
@@ -0,0 +1,35 @@
+#include <test/unit/math/test_ad.hpp>
+#include <limits>
+
+template <typename T, typename U>
+typename boost::math::tools::promote_args<T, U>::type g1(const T& x,
```

Comment:
It might be better to just use the Stan Math type trait `return_type_t` here.

return_type_t<T,U> g1(const T& x,...

Same goes for the other two uses of promote_args in this file.

Reply:
changed type metaprogram